vpdq/setup.py

```diff
@@ -1,40 +1,34 @@
 from setuptools import setup
 from setuptools.extension import Extension
-from setuptools.command.build_py import build_py
+from setuptools.command.build_ext import build_ext                                     
 import sys
 import subprocess
 import os
 import numpy
 from pathlib import Path
 
-read_me = Path.cwd() / Path("README.md")
+DIR = Path(__file__).parent
+read_me = DIR / Path("vpdq/README.md")
 long_description = read_me.read_text()
+version = (DIR / "vpdq/version.txt").read_text(encoding="utf-8").strip()
 
-
-class Build(build_py):
-    """Customized setuptools build command - builds protos on build."""
-
+class build_ext(build_ext):
     def run(self):
-        os.chdir("cpp")
-        command = ["mkdir", "build"]
-        subprocess.call(command)
-        os.chdir("build")
-        command = ["cmake", ".."]
-        if subprocess.call(command) != 0:
-            sys.exit(-1)
         command = ["make"]
-        if subprocess.call(command) != 0:
-            sys.exit(-1)
-        os.chdir("../../")
-        build_py.run(self)
-
+        try:
+            subprocess.check_call(command, cwd="vpdq/cpp")
```

Comment:
blocking: Using a non-absolute CWD here still depends on the current working directory, which means you'll get different results based on different commands.

As part of your test plan, can you invoke the script from two different directories, and make sure it works in both?